python/examples/ring.py

```diff
@@ -0,0 +1,57 @@
+# Calculating 2d ring-resonator modes, from the Meep tutorial.
+from __future__ import division
+
+import sys
+
+import meep as mp
+from meep.geom import Cylinder, epsilon, Medium, Vector3
+from meep.source import GaussianSource, Source
+from meep.simulation import Simulation, Mirror, Pml, no_size
+
+
+def main(args):
+    n = 3.4  # index of waveguide
+    w = 1  # width of waveguide
+    r = 1  # inner radius of ring
+    pad = 4  # padding between waveguide and edge of PML
+    dpml = 2  # thickness of PML
+    sxy = 2 * (r + w + pad + dpml)  # cell size
+
+    # Create a ring waveguide by two overlapping cylinders - later objects
+    # take precedence over earlier objects, so we put the outer cylinder first.
+    # and the inner (air) cylinder second.
+    dielectric = Medium(epsilon_diag=epsilon(n))
```

Comment:
My `epsilon` function in `geom.py` squared the argument passed to it, which is why this worked.  I've aligned it with scheme now.